akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -98,15 +106,34 @@ class Http2ServerDemux extends GraphStage[BidiShape[Http2SubStream, FrameEvent,
         bufferedFrameOut.push(SettingsFrame(Nil)) // server side connection preface
       }
 
-      var incomingStreams = Map.empty[Int, SubStream]
+      var closedAfter: Option[Int] = None
+      var incomingStreams = mutable.Map.empty[Int, SubStream]
       var totalOutboundWindowLeft = Http2Protocol.InitialWindowSize
       var streamLevelWindow = Http2Protocol.InitialWindowSize
 
+      def lastStreamId: Int = {
+        incomingStreams.lastOption.map(_._1).getOrElse(0)
+      }
+
+      def goAwayFrame(errorCode: ErrorCode = Http2Protocol.ErrorCode.PROTOCOL_ERROR): Unit = {
```

Comment:
Can we call it `pushGOAWAY`? I really liked that convention we had in tests about the frame types when we push/pull etc. Here it'd also help to know it performs a push.